File: typescript/1684.CountTheNumberOfConsistentStrings.ts
function countConsistentStrings(allowed: string, words: string[]): number {
  /**
   * Convert the allowed string to a set for O(1) lookup
   * Use reduce to count the number of words that are consistent
   */
  let allowed_set: Set<string> = new Set(allowed);
  return words.reduce((count, word) => {
    // check if every character in the word is in the allowed set
    if (Array.from(word).every((char) => allowed_set.has(char))) 
      {return count + 1;}
    return count;
  }, 0);
}

File: python/easy/1684-count-the-number-of-consistent-strings.py
from typing import List


class Solution:
    def countConsistentStrings(self, allowed: str, words: List[str]) -> int:
        # convert allowed to a set for O(1) lookup
        allowed_set = set(allowed)
        # count the number of words that are consistent
        return sum(all(char in allowed_set for char in word) for word in words)
